src/typescript/frontend/src/utils/ans.ts

```diff
@@ -0,0 +1,105 @@
+import { APTOS_NETWORK } from "lib/env";
+
+function getRequestURL(address: string, isPrimary: boolean) {
+  return isPrimary
+    ? `https://www.aptosnames.com/api/${APTOS_NETWORK}/v1/primary-name/${address}`
+    : `https://www.aptosnames.com/api/${APTOS_NETWORK}/v1/name/${address}`;
+}
+
+export const LOCALSTORAGE_ANS_KEY = "ans-names";
+export const ANS_CACHE_TIME = 60 * 60 * 24 * 7 * 1000; // One week
+export const MAX_ITEMS_IN_CACHE = 3000;
+
+type ANSMap = Map<string, { name: string | null; isNull: boolean; expiry: number }>;
+type ANSPromiseMap = Map<string, Promise<{ name: string | null; isNull: boolean; expiry: number }>>;
+
+function serializeMap(map: ANSMap) {
+  let array = Array.from(map.entries()).filter((a) => a[1].expiry > new Date().getTime());
+  if (array.length > MAX_ITEMS_IN_CACHE) {
+    array = array
+      .sort((a, b) => a[1].expiry - b[1].expiry)
+      .slice(array.length - MAX_ITEMS_IN_CACHE);
+  }
+  return JSON.stringify(array);
+}
```

Comment:
```suggestion
type ANSMapValue = { name: string | null; isNull: boolean; expiry: number };
type ANSMap = Map<string, ANSMapValue>;
type ANSPromiseMap = Map<string, Promise<ANSMapValue>>;

function serializeMap(map: ANSMap) {
  const now = new Date().getTime();
  const notExpired = (entry: [string, ANSMapValue]) => entry[1].expiry > now;

  const res = Array.from(map.entries())
    .filter(notExpired)
    .sort((a, b) => a[1].expiry - b[1].expiry)
    .slice(0, MAX_ITEMS_IN_CACHE);
    
  return JSON.stringify(res);
}
```

Can clean this up a little bit, feel free to use this or change it or whatever, but making it a bit more readable is nice

You might need to add a `//prettier-ignore` above the `const res = ...` line. I personally think the multiple chained function calls on a single line is nearly unreadable so I always try to format it as multiple lines

Reply:
Also double check my logic on this, free-handed it in the github comment